src/orbit-db-address.js

```diff
@@ -14,6 +14,10 @@ class OrbitDBAddress {
     return OrbitDBAddress.join(this.root, this.path)
   }
 
+  toJSON () {
```

Comment:
I'm afraid having a toJSON method that returns a string makes me a little dizzy. Can we check the type in `address.parse` and handle it accordingly there?

Reply:
Thank you for taking time to look at this.
I thought about changing the parser initially when wanting to make this pr but decided not to.
I am opposed to only changing the parser and not having toJSON return a string.
One reason for this is I think orbitdb addresses are better described/more readable as strings. They also resemble multiaddrs a bit which have a toJSON method that return a string here: https://github.com/multiformats/js-multiaddr/blob/072326e4bc9a0fa5fab0316d8160d2b493534e8f/src/index.js#L69
It seems like most things in javascript that have a toString method would also be a string in json.
I'm not opposed to changing the parser to handle the current json form of the address but I think its unnecessary.

Reply:
Thanks for making the PR!

I suppose one could make the the argument that a string in JS could feasibly be viewed as "valid JSON", I still don't like the idea of messing with expectations in this way. JS types are more are less free-for all, but I'd like to honor them as much as possible... Also, if `JSON.parse("xxxx")` didn't throw an error and returned a string I might be more persuaded.

Other people are free to weigh in (maybe toss it out to Gitter?) but for better or worse I'm still on the side of moving this handler to the parse function. I'm sorry if this is frustrating.

Reply:
Its not frustrating, I appreciate the honest feedback. <3

Reply:
it is possible to parse json strings like so `JSON.parse('"asdf"')`

Reply:
I'm sorry I didn't ask this before - what exactly is the error you're getting and on which line?

Reply:
```
address.constructor.parse(
  JSON.parse(JSON.stringify(address))
)

resulting error:
orbit-db-address.js:42 Uncaught Error: Not a valid OrbitDB address: [object Object]
    at Function.parse (orbit-db-address.js:42)
    at <anonymous>:1:18
```
https://github.com/orbitdb/orbit-db/blob/ed697fdfd6bead8752d6d0a4760270653d3215a8/src/orbit-db-address.js#L30
here `parts` is equal to `["[object Object]"]` which makes `accessControllerHash` conditional false.

Reply:
Can we start by writing a failing test to catch this case? And then I can play around with it a little bit on my own and provide more cohesive thoughts.

Reply:
I did include a test that sounds like what you want, but if its not lmk.

Reply:
Ah, so you did! I'll take a look at this after I've had some more sleep :)